### ISSUE TITLE
[java] UnnecessaryImport false positive for on-demand imports in JavaDoc

### ISSUE THREAD

**Issue by kris-scheibe** [pmd/pmd#2304]
**Affects PMD Version:**
6.21+
7.0.0

**Rule:** [UnnecessaryImport](https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#unnecessaryimport)

**Description:**
When an on-demand import is only used in a JavaDoc `@link`, `@see` or similar tag, the import is flagged.

This rule requires type-resolution to match the on-demand import, which isn't performed for JavaDoc content (I think...)

I already played around with this a little bit, but my solution seems like a hack, as I'm not sure, if the internal APIs were meant to be used that way and I feel that this might break easily if the code-resolution logic changes: kris-scheibe/pmd@3b195323f2402dde9ca2e72aca66f925d826bd38

Do you guys have any ideas or comments?

**Code Sample demonstrating the issue:**

```java
import java.util.*;

public class Foo {
    /**
     * {@link  ArrayList  arraylist}
     */
    public void test() {}
}
```

**Running PMD through:** *Other*
